app/src/main/java/com/quarkworks/apartmentgroceries/service/SyncGroceryItem.java

```diff
@@ -38,13 +48,30 @@ public void done(@Nullable JSONObject jsonObject) {
                 try {
 
                     JSONArray groceryJsonArray = jsonObject.optJSONArray(JsonKeys.RESULTS);
-
-                    for (int i = 0; i < groceryJsonArray.length(); i++) {
-                        RGroceryItem groceryItem = realm.createObject(RGroceryItem.class);
-                        groceryItem.setName(groceryJsonArray.getJSONObject(i).optString(JsonKeys.NAME));
+                    if (groceryJsonArray != null) {
+                        for (int i = 0; i < groceryJsonArray.length(); i++) {
+                            RGroceryItem groceryItem = realm.createObject(RGroceryItem.class);
+                            groceryItem.setGroceryId(groceryJsonArray.getJSONObject(i)
+                                    .getString(JsonKeys.OBJECTID));
+                            groceryItem.setName(groceryJsonArray.getJSONObject(i)
+                                    .getString(JsonKeys.NAME));
+                            groceryItem.setGroupId(groceryJsonArray.getJSONObject(i)
+                                    .getJSONObject(JsonKeys.GROUPID).getString(JsonKeys.OBJECTID));
+                            groceryItem.setGroupName(groceryJsonArray.getJSONObject(i)
+                                    .getJSONObject(JsonKeys.GROUPID).getString(JsonKeys.NAME));
+                            groceryItem.setCreatedBy(groceryJsonArray.getJSONObject(i)
+                                    .getJSONObject(JsonKeys.CREATEDBY).getString(JsonKeys.USERNAME));
+                            JSONObject purchasedByObj = groceryJsonArray.getJSONObject(i)
+                                    .optJSONObject(JsonKeys.PURCHASEDBY);
+                            if (purchasedByObj != null) {
+                                groceryItem.setPurchasedBy(purchasedByObj.getString(JsonKeys.USERNAME));
+                            }
+                        }
+
+                        realm.commitTransaction();
+                    } else {
+                        realm.cancelTransaction();
```

Comment:
I think this code would work better if you use two try-catches. One should be for getting the JSONArray and the other should be for parsing each grocery item individually. We don't want it to cancel the transaction if there is a JSONException in one grocery item, we just don't want to save that grocery item.